src/MAGMA.jl

```diff
@@ -23,25 +23,150 @@ const MagmaSomeVec       = 305
 const MagmaOverwriteVec  = 306
 const MagmaBacktransVec  = 307
 
+# https://github.com/mweastwood/MAGMA.jl
+
+export device_reset,
+       set_vector, set_matrix, get,
+       axpy!, dot,
+       gemv!
+
+import Base: length, size
+
+
+# some types' definitions and their accordingly chararcters
+const magmaTypes = Union{Float32, Float64, ComplexF32, ComplexF64}
+const magmaTypesList =  (Float32,Float64,ComplexF32,ComplexF64)
+const typeChar = Dict(
+    Float32 => 's',
+    Float64 => 'd',
+    ComplexF32 => 'c',
+    ComplexF64 => 'z'
+)
+
 """
 the path to magma binary
 """
 const libmagma = "/usr/local/magma/lib/libmagma.so"
 
+
+# https://github.com/mweastwood/MAGMA.jl
+struct set_vector_struct{T <: magmaTypes}
```

Comment:
why there is still a custom type for this? why builtin `Array` cannot be converted to pointer?

Reply:
And again don't use snake case for types.